testing/trino-testing/src/main/java/io/trino/testing/BaseFailureRecoveryTest.java

```diff
@@ -74,12 +75,15 @@
     protected static final int INVOCATION_COUNT = 1;
     private static final Duration MAX_ERROR_DURATION = new Duration(5, SECONDS);
     private static final Duration REQUEST_TIMEOUT = new Duration(5, SECONDS);
+    private static final int MAX_PARALLEL_TEST_CONCURRENCY = 4;
 
     private final RetryPolicy retryPolicy;
+    private final Semaphore parallelTestsSemaphore;
```

Comment:
Static vs instance.

If we have two classes within a suite, we should decide whether we want total limit, or per-class limit

(I am not advocating in any direction)

Reply:
Yeah - I know we have two options. I went with per-instance without any strong feeling that it is a better way.